packages/ui-components/src/components/tooltip/tooltip.tsx

```diff
@@ -44,25 +48,18 @@ export class KvTooltip {
 		const child = this.el.shadowRoot.querySelector('#content');
 		const tooltip = this.el.shadowRoot.querySelector('#tooltip') as HTMLElement;
 
-		const position = isEmpty(this.allowedPositions) ? this.position : undefined;
-		const middleware = [offset(DEFAULT_OFFSET), shift(DEFAULT_SHIFT_CONFIG)];
-
-		if (isEmpty(position)) {
-			middleware.push(autoPlacement({ ...DEFAULT_AUTOPLACEMENT_CONFIG, allowedPlacements: this.allowedPositions }));
+		if (tooltip === null) {
+			return;
 		}
 
-		function update() {
-			computePosition(child, tooltip, {
-				...DEFAULT_TOOLTIP_CONFIG,
-				placement: position,
-				middleware
-			}).then(({ x, y }) => {
+		const update = () => {
+			computePosition(child, tooltip, this.getOptions()).then(({ x, y }) => {
 				Object.assign(tooltip.style, {
```

Comment:
do we really need to use Object.assign or can we mutate the values directly?

Reply:
Can be mutated directly 👍